src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,348 @@
+      )}
+      {counterLimit && (
+        <LabelSmall
+          weight="Regular"
+          color="grey-700"
+        >{`${valueCount} / ${counterLimit}`}</LabelSmall>
+      )}
+    </View>
+  );
+};
+
+export const TextInputBase = ({
+  disabled = false,
+  placeholder,
+  value = "",
+  icon,
+  counterLimit,
+  helperText,
+  onChangeText,
+  onValidate,
+  secretInput,
+  errorMessage
+}: InputTextProps) => {
+  const labelSharedValue = useSharedValue(0);
+  const [inputStatus, setInputStatus] = React.useState<InputStatus>(
+    disabled ? "disabled" : "initial"
+  );
+  const [isValid, setIsValid] = React.useState<boolean | undefined>(undefined);
+  const [isSecretInput, setSecretInput] = React.useState<boolean>(
+    secretInput ?? false
+  );
+
+  const inputRef = useRef<TextInput>(null);
+
+  const boxStyle: ViewStyle = useMemo(() => {
+    if (inputStatus === "focused") {
+      return {
+        borderColor: IOColors["blueIO-500"],
+        borderWidth: 2
+      };
+    }
+    if (inputStatus === "error") {
+      return {
+        borderColor: IOColors["error-600"],
+        borderWidth: 1
+      };
+    }
+    return {
+      borderColor: IOColors["grey-200"],
+      borderWidth: 1
+    };
+  }, [inputStatus]);
+
+  const animatedLabelProps = useAnimatedStyle(() => ({
+    fontSize: withTiming(interpolate(labelSharedValue.value, [0, 1], [16, 12])),
+    top: withTiming(interpolate(labelSharedValue.value, [0, 1], [0, -16]))
+  }));
+
+  useEffect(() => {
+    if (value.length > 0) {
+      labelSharedValue.value = 1;
+    } else {
+      if (inputStatus !== "focused") {
+        labelSharedValue.value = 0;
+      }
+    }
+  }, [labelSharedValue, value, inputStatus]);
+
+  const onTextInputPress = () => {
+    if (disabled) {
+      return;
+    }
+    labelSharedValue.value = 1;
+    setInputStatus("focused");
+    inputRef?.current?.focus();
+  };
+
+  const onChangeTextHandler = useCallback(
+    (text: string) => {
+      if (counterLimit && text.length > counterLimit) {
+        return;
+      }
+      onChangeText(text);
+    },
+    [counterLimit, onChangeText]
+  );
+
+  const onBlurHandler = useCallback(() => {
+    if (!value) {
+      labelSharedValue.value = 0;
+    }
+    if (secretInput) {
+      setSecretInput(true);
+    }
+    if (onValidate) {
+      const isValid = onValidate(value);
+      setIsValid(isValid);
+      if (!isValid) {
+        setInputStatus("error");
+        triggerHaptic("notificationError");
+        return;
+      } else {
+        triggerHaptic("notificationSuccess");
+      }
+    }
+    setInputStatus("initial");
+  }, [onValidate, value, labelSharedValue, secretInput]);
+
+  return (
+    <>
+      <Pressable
+        onPress={onTextInputPress}
+        style={[
+          inputStatus === "disabled" ? { opacity: 0.5 } : {},
+          boxStyle,
+          styles.textInput
+        ]}
+      >
+        {icon && (
+          <>
+            <Icon name={icon} size={24} />
```

Comment:
According to the design guidelines, the icon color is not dark:
```suggestion
            <Icon name={icon} color="grey-300" size={24} />
```